### PR TITLE
feat: support automatic redirection to __sys__

### PR DESCRIPTION
**Pull request by NoyException**

### CODE REVIEW

File: catalog/internal_tables.go [hunk 1]
```diff
@@ -287,6 +288,29 @@ var InternalTables = struct {
 		},
 		DDL: "pid INTEGER PRIMARY KEY, usesysid TEXT, usename TEXT, application_name TEXT, client_addr TEXT, client_hostname TEXT, client_port INTEGER, backend_start TIMESTAMP, backend_xmin INTEGER, state TEXT, sent_lsn TEXT, write_lsn TEXT, flush_lsn TEXT, replay_lsn TEXT, write_lag INTERVAL, flush_lag INTERVAL, replay_lag INTERVAL, sync_priority INTEGER, sync_state TEXT, reply_time TIMESTAMP",
 	},
+	PGRange: InternalTable{
+		Schema:       "__sys__",
+		Name:         "pg_range",
+		KeyColumns:   []string{"rngtypid"},
+		ValueColumns: []string{"rngsubtype", "rngmultitypid", "rngcollation", "rngsubopc", "rngcanonical", "rngsubdiff"},
+		DDL:          "rngtypid TEXT PRIMARY KEY, rngsubtype TEXT, rngmultitypid TEXT, rngcollation TEXT, rngsubopc TEXT, rngcanonical TEXT, rngsubdiff TEXT",
+		//COPY pg_catalog.pg_range (rngtypid, rngsubtype, rngmultitypid, rngcollation, rngsubopc, rngcanonical, rngsubdiff) FROM stdin;
```
Comment (TianyuZhang1214) [inline catalog/internal_tables.go:297]: Remote the useless comment here?

**Review by TianyuZhang1214 — CHANGES_REQUESTED** [review]

File: pgserver/stmt.go [hunk 2]
```diff
@@ -259,3 +261,24 @@ func RemoveComments(query string) string {
 
 	return buf.String()
 }
+
+// precompile a regex to match any "from pg_catalog.pg_stat_replication" in the query.
+var pgCatalogRegex *regexp.Regexp
+
+func getPgCatalogRegex() *regexp.Regexp {
```
Comment (fanyang01) [inline pgserver/stmt.go:268]: Not concurrent-safe. You may need `sync.Once*`: https://pkg.go.dev/sync#OnceValue

File: pgserver/stmt.go [hunk 3]
```diff
@@ -259,3 +261,24 @@ func RemoveComments(query string) string {
 
 	return buf.String()
 }
+
+// precompile a regex to match any "from pg_catalog.pg_stat_replication" in the query.
```
Comment (fanyang01) [inline pgserver/stmt.go:265]: This comment needs to be updated.

File: pgserver/stmt.go (hunk 2 above)
Comment (fanyang01) [reply pgserver/stmt.go:273]: https://pkg.go.dev/regexp#Regexp

> A Regexp is safe for concurrent use by multiple goroutines, except for configuration methods, such as [Regexp.Longest](https://pkg.go.dev/regexp#Regexp.Longest).

So the RWLock is unneeded.

File: pgserver/stmt.go [hunk 4]
```diff
@@ -259,3 +262,27 @@ func RemoveComments(query string) string {
...
+func getPgCatalogRegex() *regexp.Regexp {
+	initPgCatalogRegex.Do(func() {
+		var tableNames []string
+		for _, table := range catalog.GetInternalTables() {
+			if table.Schema != "__sys__" {
+				continue
+			}
+			tableNames = append(tableNames, table.Name)
+		}
+		pgCatalogRegex = regexp.MustCompile(`(?i)\s+(?:pg_catalog\.)?(` + strings.Join(tableNames, "|") + `)`)
```
Comment (fanyang01) [inline pgserver/stmt.go:281]: Maybe we need to prefix the pattern with `(FROM|JOIN|,)` to avoid undesired replacement.

**Review by fanyang01 — APPROVED** [review]
Thanks!

**Review by TianyuZhang1214 — APPROVED** [review]
LGTM

**Pull request merged**